blocks/library/image/style.scss

```diff
@@ -9,17 +9,28 @@
 	&[data-align="left"] {
 		float: left;
 		margin-right: $block-padding;
+		@include break-medium() {
+			margin-left: calc( 50% - #{ $visual-editor-max-width / 2 } );
+		}
 	}
 
 	&[data-align="right"] {
 		float: right;
 		margin-left: $block-padding;
+		@include break-medium() {
+			margin-right: calc( 50% - #{ $visual-editor-max-width / 2 } );
+		}
 	}
 
 	&[data-align="wide"] {
+		max-width: 1100px;
+	}
+
+	&[data-align="full"] {
+		max-width: 100%;
 		padding-left: 0;
 		padding-right: 0;
-		margin-right: -#{ $block-padding + $block-mover-margin };	/* Compensate for .editor-visual-editor centering padding */
+		box-sizing: border-box;
```

Comment:
Shouldn't need this:

https://github.com/WordPress/gutenberg/blob/73bd4ab/editor/assets/stylesheets/main.scss#L17-L20